packages/drivers/odsp-driver/src/odspUtils.ts

```diff
@@ -307,12 +313,16 @@ export function toInstrumentedOdspTokenFetcher(
                 }
                 return token;
             }, (error) => {
+                // There is an important but unofficial contract here where token providers can set canRetry: true
+                // to hook into the driver's retry logic (e.g. the retry loop when initiating a connection)
+                const rawCanRetry = error?.canRetry;
                 const tokenError = wrapError(
                     error,
-                    (errorMessage) => new NonRetryableError(
+                    (errorMessage) => new NetworkErrorBasic(
```

Comment:
Rest seems fine. Is NetworkErrorBasic the best one to use here, since network is not involved? Should we create Retryable/NonRetryable error object based on the canRetry condition which you are adding instead of rawCanRetry ?

Reply:
If you look at NetworkErrorBasic, you'll see it doesn't actually have an opinion on the errorType or anything. Probably should be named DriverErrorBasic or something. No inclined to rename it as part of this PR though. 

And I don't understand your comment about canRetry v rawCanRetry, can you explain again?